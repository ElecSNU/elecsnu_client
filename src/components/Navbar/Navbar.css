#navbar {
    height: 7.5%;
    display: flex;
    background-color: var(--accent1);
    justify-content: space-between;
    align-items: center;
    padding: 0 10%;
    z-index: 1;
}

#navbar a {
    color: var(--light);
    font-size: 1.5rem;
    transition: color 0.6s;
}

#navbar a:hover {
    color: var(--dark);
}

.brand a {
    font-size: 2.5rem;
}

.links {
    width: 50%;
    display: flex;
    justify-content: space-evenly;
}
