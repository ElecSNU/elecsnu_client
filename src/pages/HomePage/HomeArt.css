#home {
    height: 100%;
    display: flex;
    align-items: center;
    overflow: hidden !important;
}

#home .voting-banner {
    margin: 0 5rem;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
}

#home .voting-banner .submit-button {
    margin-top: 2rem;
    width: 100%;
}

#home .home-art {
    width: 100%;
    height: 100%;
    position: relative;
    overflow: hidden;
}

#home .home-art img {
    width: 120%;
    position: absolute;
    right: -25%;
    bottom: 0;
}
