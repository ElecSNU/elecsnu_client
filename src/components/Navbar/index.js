import React, { useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import { useStoreActions, useStoreState } from 'easy-peasy';

import './Navbar.css';

const Navbar = () => {
    const logoutAction = useStoreActions(
        (actions) => actions.accountModel.logout
    );

    const loggedIn = useStoreState(
        (store) => store.accountModel.user_logged_in
    );

    let userLoggedIn = window.localStorage.getItem('token');

    useEffect(() => {
        // eslint-disable-next-line react-hooks/exhaustive-deps
        userLoggedIn = window.localStorage.getItem('token');

        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [loggedIn]);

    return (
        <div data-testid = "navbar" id='navbar'>
            <div className='brand'>
                <NavLink
                    
                    className='normal-text logdata-testid = "ElecSNU"out'
                    to='/'
                >
                    ElecSNU
                </NavLink>
            </div>
            <div className='links'>
                <NavLink
                    className='normal-text'
                    to='/features'
                >
                    Features
                </NavLink>
                <NavLink
                    className='normal-text'
                    to='/about-us'
                >
                    About Us
                </NavLink>
                {userLoggedIn ? (
                    <NavLink
                        className='normal-text'
                        to='/dashboard'
                    >
                        Dashboard
                    </NavLink>
                ) : (
                    <NavLink
                        className='normal-text'
                        to='/login'
                    >
                        Login
                    </NavLink>
                )}
                {userLoggedIn && (
                    <NavLink
                        className='normal-text logout'
                        to='/'
                        onClick={() => logoutAction()}
                    >
                        Logout
                    </NavLink>
                )}
            </div>
        </div>
    );
};

export default Navbar;
