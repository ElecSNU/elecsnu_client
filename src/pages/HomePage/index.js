import React from 'react';

import './HomeArt.css';

import homeArt from '../../assets/images/home-art.svg';
import sc1 from '../../assets/images/scroll_1.png';
import sc2 from '../../assets/images/scroll_2.png';
// import gfont from 'https://fonts.googleapis.com/css?family=Montserrat' 

const HomePage = () => {
   
    return (
        <>
            <div
                id='home'
                style={{
                    width: '100%',
                    height: '100vh',
                    backgroundColor: '#004aad'
                }}
            >
                <div className='voting-banner' >
                        <h1  style={{
                        // fontFamily: "Montserrat"
                    }}>LET'S MAKE VOTING FUN!</h1>
                        <br/>
                        <h4 >
                            A step towards secure and intuitive digital voting 
                        </h4>

                </div>
                <div className='home-art'>
                    <img src={homeArt} alt='home-art' />
                </div>
            </div>
            {/* <div className='sc1'> */}
                <h1>All about online voting systems</h1>     
            {/* </div>s */}
            <div className='sc1'>
                <img src={sc1} alt='sc1' />    
            </div>
            <div className='sc1'>
                <img src={sc2} alt='sc1' />    
            </div>
            
        </>
    );
};

export default HomePage;
