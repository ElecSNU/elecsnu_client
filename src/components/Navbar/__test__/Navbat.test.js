import { render, screen, fireEvent} from '@testing-library/react';
import Navbar from './../index';
import ReactDOM from 'react-dom'
import React from 'react'
import renderer from 'react-test-renderer'
import HomePage from './../../../pages/HomePage/index';
import { useStoreActions, useStoreState } from 'easy-peasy';

test('renders the without crashing', () => {
    const div = document.createElement("div")
    ReactDOM.render(<index></index>, div)
});

// test('opens the ElecSNU page',()=>{
//     // render(<Navbar></Navbar>);
//     const div = document.createElement("div")
//     const tree = renderer.create(<index></index>).toJSON();
//     expect(tree).toMatchSnapshot();
//     fireEvent.click(getByTestId('ElecSNU'))
//     expect(tree).toMatchSnapshot();

// });

// describe("<NavLink  />", () => {
//     it("1.contains correct passed prop", () => {
//         const comp = (
//             <NavLink
//                     className='normal-text logout'
//                     to='/'
//                 >
//                     ElecSNU
//                 </NavLink>
//         );
//         const wrapper = shallow( comp );
//         expect(wrapper.instance().prop().to).to.equal("/");
//     });
// });    