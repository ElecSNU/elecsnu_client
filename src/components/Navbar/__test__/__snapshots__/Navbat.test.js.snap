// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opens the ElecSNU page 1`] = `<index />`;
